### DISCUSSION TITLE
internal error durring connect

### DISCUSSION THREAD

**Discussion by kenjimomofuku** [Guilouz/Klipper-Flsun-Speeder-Pad#172] [Q&A]
speeder pad freshly reflashed in 1.2, followed by the wiki :

passwrd: ok
timezone first crash after cmd [timedatectl] I'm still in the states
wifi-location, after the manipulation (the command gives the right country)
Update dependencies: now it gets really messy: I get the Network error: after the interminable freeze on ‘Processing triggers for systemd’. I unplug-plug, menu>network and then it's IMPOSSIBLE to reconnect, it runs 2/3 times without success
I reflash the pad in 1.4, the error message remains: ‘internal error durring connect: can't convert negative number to unsigned.’
However, it is possible to connect to the network. But impossible to ssh, mcu not recognized.
I've redone the process 3 times, having to reflash to 1.2 each time to recover the ssh. I'm sure I followed the steps in the right order, but as it stands, v400 is garbage...

So I wouldn't mind a helping hand

I'd also like to make a suggestion: a disclamer in the github, which explains that the modification allows more freedom but can also pose huge problems of stability, printing, immobilising the printer for days etc... in short being risky and complex.

**Comment by ven001** [comment]
salut Cyril j'ai le même problème pour me reconnecter après la commande "sudo apt-get install network-manager -y". j'ai bien débranché et re branché le pad mais pas de message d'erreur sur klipperscreen. et quand j'essai de me reconnecter a ma box j,ai ce message qui apparait 
CTRL-EVENT-DISCONNECTED bssid =00:06::91:de:8f:c0 reason=1 locally_generated=1
CTRL-EVENT-REGDOM-CHANGE init=CORE type=world
CTRL-EVENT-REGDOM-CHANGE init=USER type=COUNTRY alpha2 = FR
merci a toi

**Reply by kenjimomofuku** [reply]
Idem je crois

**Reply by kenjimomofuku** [reply]
dans çe cas essaye de remettre le firmware d'origine, avant change de pts d'accès ça peut potentiellement marcher

**Comment by kennyb08** [comment]
je ne suis pas le seul. sa fait plus de 2 semaines que j'ai le soucie et aucune réponse...

**Reply by kenjimomofuku** [reply]
Oui c'est la cata, guilouz aurait réussi après une dizaine de tentatives, mais c'est tout de même très tangible, du coté de flsun le git semble aussi s'être épaissît... bref la v400 et deja pas forcément très fiable, mais si faire une maj c'est jouer avec le feu ça devient complexe...

**Reply by kennyb08** [reply]
En attendant des nouvelles je vais réinstaller la version d'origine de Flsun car la je ne peux plus rien faire

**Reply by kenjimomofuku** [reply]
Tu pourras me dire si le firmware d'origine s'installe ? Je n'ai pas pu le faire fonctionner personnellement

**Reply by kennyb08** [reply]
Problème après l'installation impossible de le lancer 
![FB_IMG_1718715592451](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/62797512/f1745fa0-c710-4dbe-8e13-2864da9962e7)

**Reply by kenjimomofuku** [reply]
j'avais exactement pareil

**Reply by kenjimomofuku** [reply]
ok j'ai parlé un peu vite finalement j'ai pu remettre le firmware d'origine (enfin ça ressemble), en gros le code erreur viendrait d'un problème avec le mcu
 tu trouvera dans ce git : https://github.com/Flsun3d/Flsun_Speeder_Pad?tab=readme-ov-file#upload-configuration-to-speeder-pad
- un firmware 1.2 un peu diffèrent du v600 que j'avais essayé
- l'image à flash dans la carte mère (robin35 perso)

avec ça j'ai pu me connecter et faire bouger la tête

**Reply by SckyzO** [reply]
Bah après l'update tu as suivis le reste du tuto ? 
Tu flash, Tu installes Klipper, puis tu Met à jour la Motherboard avec la nouvelle version de klipper
Tu as bien l'accès au SSH / Network ? 
Dans le wiki tu suis toutes les étapes du menu les unes après les autres ... tout simplement. 
J'ai fait l'installation plusieurs fois, jamais aucun problème.

**Reply by kenjimomofuku** [reply]
Non justement pendant l'update avec la version de guilouz, après la commande `sudo apt-get install network-manager -y` je perd toute possibilité de ssh. Donc je ne peut pas "tout simplement" suivre les étapes. J'ai  fait l'Install Guilouz à plusieurs reprise dans le passé avec obico, mobileraker,  neopixel, adxl et tout le train train, Mais là impossible sur mon pad... D'ailleur à vrai dire j'ai eu tellllllement de problème d'impression, de webcam, de maj... sur ce firmware qu'il parait sage d'abandonner le bidouillage

**Comment by Guilouz** [comment]
Je ne peux faire plus que cela, j'ai restauré mon pad de test des dizaines de fois et j'ai suivi la procédure plusieurs fois sans problème, à chaque fois j'ai pu me reconnecter.
Le problème étant network-manager qui fait freezer le pad lors de l'installation, je n'ai trouvé aucune solution à cela pour éviter de reboot le pad et se reconnecter au Wifi pour poursuivre.

D'ailleurs d'autres personnes ont également réussi à faire l'installation.

**Reply by kenjimomofuku** [reply]
Ok mon problème viendrait donc de la re-connexion, je vais essayer de chercher comment résoudre cet erreur ce soir mais comme je perd le ssh je devrais trouver avant de lancer l'installation...
![image](https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/assets/171831831/47218773-cc23-4e76-9da3-b932e90b6a36)
Je tiens informé au cas où

**Reply by kennyb08** [reply]
> Ok mon problème viendrait donc de la re-connexion, je vais essayer de chercher comment résoudre cet erreur ce soir mais comme je perd le ssh je devrais trouver avant de lancer l'installation... ![image](https://private-user-images.githubusercontent.com/171831831/340651911-47218773-cc23-4e76-9da3-b932e90b6a36.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg3MDk2MDUsIm5iZiI6MTcxODcwOTMwNSwicGF0aCI6Ii8xNzE4MzE4MzEvMzQwNjUxOTExLTQ3MjE4NzczLWNjMjMtNGU3Ni05ZGEzLWI5MzJlOTBiNmEzNi5qcGVnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYxOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MThUMTExNTA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2I5N2EzNjhhOWFkZTNkYTk2ZjQxOTM5MGEzNzNiYzVmMTc2YTVhMGQ2YjFjZmY5Y2FhMzJhZjY3NmZlZGI3MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.yB9JP-4dcmbn6-XxhyD4VgePfi5gdiDKOsi3cfYeAMw) Je tiens informé au cas où

Perso j'ai essayé même réinitialiser ma box pensant que sa venait d'elle mais rien à faire impossible de mon côté

**Reply by kenjimomofuku** [reply]
Oui wifi6/5 aucun changement, peut être avec la manip d'avant ou l'on change de pays, mais je crois l'avoir fait sans, et même problème

**Reply by SckyzO** [reply]
Moi j'ai suivi la procédure, certes fastidieuse, mais fonctionne très bien. 
Cela dit j'avais des problèmes de connexions avant la mise à jour, du coup perso ce que j'ai fait, j'ai acheté un relai wifi 5 a 20€ sur amazon, je l'ai mis sur mon wifi avec son propre SSID (wifi_imprimante), et activé uniquement le 2.4Ghz. Je n'ai plus aucun soucis. Le soucis n'est pas la mise à jour mais la puce de la FLSun je pense.

**Reply by SckyzO** [reply]
> Oui wifi6/5 aucun changement, peut être avec la manip d'avant ou l'on change de pays, mais je crois l'avoir fait sans, et même problème

Et test sur une connexion uniquement en 2.4 GHz

**Reply by kenjimomofuku** [reply]
> > Oui wifi6/5 aucun changement, peut être avec la manip d'avant ou l'on change de pays, mais je crois l'avoir fait sans, et même problème
> 
> 
> 
> Et test sur une connexion uniquement en 2.4 GHz

Ça ne coûte rien d'essayer mais ça semblerait loufoque en sachant que ça marchait parfaitement avant, a tester...

**Reply by kenjimomofuku** [reply]
j'ai réussi à me connecter, sur un point d'accès, mais impossible de lancer un ssh, donc pas mieux...

**Reply by mithrandir63** [reply]
Salut a tous,

Perso j'ai suivi à la lettre la procédure de Cyril j'ai une freebox Delta où j'ai effectivement séparé les points d'accès 2.4GHz et 5 GHz = ssid différents ( je fais toujours ça sur mes routeurs ou extender) 
Y a rien de plus pénible qu'un périphérique qui n'arrête pas de switcher entre le 2,4 et le 5 GHz en fonction de son humeur.

D'expérience attention aussi à certains caractères spéciaux dans les clés wifi ça pose quasiment jamais de problème mais parfois le petit grain de sable était là.....

Après ce que je ne sais pas est-ce qu'il y a différents hardware dans les Flsun Speeder PAD c'est a dire différentes versions.
Car sinon ( c'est-à-dire qu'on a tous le même Speeder PAD) si la procédure de Cyril marche sur certains Speeder PAD alors si ça plante sur d'autres le problème vient forcément d'un autre élément extérieur....

Perso j'ai fait une installation multi instance je n'ai pas eu de problème sur la partie wifi juste quelques petits trucs à ajuster sur la fin du tutoriel mais dû à mon côté multi instances...

Yes WE Print 😁😉

**Reply by mithrandir63** [reply]
N'empêche ce qui aurait pu aider c'est d'avoir un accès type terminal ou ligne de commande sur les Speeder pad comme ça un clavier et une souris et hop on pourrait en savoir plus et faire plus de choses...
Sur un raspberry pi avec un écran tactile on peut avoir ce type de fonction. 😉
Ctrl + Alt + F1
Si j'étais pas si loin de mon SpeederPad je testerai direct 🧐🧐

**Reply by SckyzO** [reply]
> N'empêche ce qui aurait pu aider c'est d'avoir un accès type terminal ou ligne de commande sur les Speeder pad comme ça un clavier et une souris et hop on pourrait en savoir plus et faire plus de choses... Sur un raspberry pi avec un écran tactile on peut avoir ce type de fonction. 😉 Ctrl + Alt + F1 Si j'étais pas si loin de mon SpeederPad je testerai direct 🧐🧐

Ce n'est pas possible, ce n'est pas un RPi, c'est une simple board :)

**Reply by kenjimomofuku** [reply]
> Après ce que je ne sais pas est-ce qu'il y a différents hardware dans les Flsun Speeder PAD c'est a dire différentes versions.
> Car sinon ( c'est-à-dire qu'on a tous le même Speeder PAD) si la procédure de Cyril marche sur certains Speeder PAD alors si ça plante sur d'autres le problème vient forcément d'un autre élément extérieur....

Pas impossible, j'y ai pensé, si on enlève le coté logiciel avec un re-flash, network avec différent pts d'accès, l'humain en reprenant l'Install plusieurs fois (sur 20 fois quelque ligne de code, pense pas  pouvoir faire 20 fois d'oubli lol) reste le matériel, ou les ondes extra terrestre ^^
